block/manager.go

```diff
@@ -510,15 +515,15 @@ func (m *Manager) BatchRetrieveLoop(ctx context.Context) {
 					"txCount", len(res.Batch.Transactions),
 					"timestamp", res.Timestamp)
 
-				if h, err := res.Batch.Hash(); err == nil {
-					m.bq.AddBatch(BatchWithTime{Batch: res.Batch, Time: res.Timestamp})
-					if len(res.Batch.Transactions) != 0 {
-						if err := m.store.SetMetadata(ctx, LastBatchHashKey, h); err != nil {
-							m.logger.Error("error while setting last batch hash", "error", err)
-						}
-						m.lastBatchHash = h
+				m.bq.AddBatch(BatchWithTime{Batch: res.Batch, Time: res.Timestamp})
```

Comment:
do you want to do `BatchWithTimeAndBatchData`? so that when u getNextBatch in publish, you can put the corresponding batch data (after convertBatchDataToBytes) into header.DataHash field?